packages/devtools_app/lib/src/flutter/extent_delegate_list.dart

```diff
@@ -0,0 +1,464 @@
+    delegate.onLayoutDirty = markNeedsLayout;
+    _extentDelegate = delegate;
+  }
+
+  ExtentDelegate _extentDelegate;
+
+  /// Called to estimate the total scrollable extents of this object.
+  ///
+  /// Must return the total distance from the start of the child with the
+  /// earliest possible index to the end of the child with the last possible
+  /// index.
+  ///
+  /// By default, defers to [RenderSliverBoxChildManager.estimateMaxScrollOffset].
+  @protected
+  double estimateMaxScrollOffset(
+    SliverConstraints constraints, {
+    int firstIndex,
+    int lastIndex,
+    double leadingScrollOffset,
+    double trailingScrollOffset,
+  }) {
+    return childManager.estimateMaxScrollOffset(
+      constraints,
+      firstIndex: firstIndex,
+      lastIndex: lastIndex,
+      leadingScrollOffset: leadingScrollOffset,
+      trailingScrollOffset: trailingScrollOffset,
+    );
+  }
+
+  int _calculateLeadingGarbage(int firstIndex) {
+    RenderBox walker = firstChild;
+    int leadingGarbage = 0;
+    while (walker != null && indexOf(walker) < firstIndex) {
+      leadingGarbage += 1;
+      walker = childAfter(walker);
+    }
+    return leadingGarbage;
+  }
+
+  int _calculateTrailingGarbage(int targetLastIndex) {
+    RenderBox walker = lastChild;
+    int trailingGarbage = 0;
+    while (walker != null && indexOf(walker) > targetLastIndex) {
+      trailingGarbage += 1;
+      walker = childBefore(walker);
+    }
+    return trailingGarbage;
+  }
+
+  BoxConstraints buildChildConstraints(int index) {
+    final currentItemExtent = _extentDelegate.itemExtent(index);
+    return constraints.asBoxConstraints(
+      minExtent: currentItemExtent,
+      maxExtent: currentItemExtent,
+    );
+  }
+
+  // This method is is a fork of RenderSliverFixedExtentBoxAdaptor.performLayout
+  // where we defer computations about offsets to the _extendDelegate and try
+  // to avoid logic only applicable if all children have the same extent.
+  @override
+  void performLayout() {
+    childManager.didStartLayout();
+    childManager.setDidUnderflow(false);
+
+    final double scrollOffset =
+        constraints.scrollOffset + constraints.cacheOrigin;
+    assert(scrollOffset >= 0.0);
+    final double remainingExtent = constraints.remainingCacheExtent;
+    assert(remainingExtent >= 0.0);
+    final double targetEndScrollOffset = scrollOffset + remainingExtent;
+
+    final int firstIndex =
+        _extentDelegate.minChildIndexForScrollOffset(scrollOffset);
+    final int targetLastIndex = targetEndScrollOffset.isFinite
+        ? _extentDelegate.maxChildIndexForScrollOffset(targetEndScrollOffset)
+        : null;
+
+    if (firstChild != null) {
+      final int leadingGarbage = _calculateLeadingGarbage(firstIndex);
+      final int trailingGarbage = _calculateTrailingGarbage(targetLastIndex);
+      collectGarbage(leadingGarbage, trailingGarbage);
+    } else {
+      collectGarbage(0, 0);
+    }
+
+    if (firstChild == null) {
```

Comment:
can we put all of this in the previous else statement?

Reply:
done. Also removed the spurious trailing return.